### PR TITLE
feat(joins): enable limiting joins again

### PR DESCRIPTION
**Pull request by oetyng**

### CODE REVIEW

File: src/capacity/rate_limit.rs [hunk 1]
```diff
@@ -37,20 +37,20 @@ impl RateLimit {
         RateLimit::rate_limit(bytes, full_nodes, all_nodes, prefix_len)
     }
 
-    ///
+    /// Adds this node to the list o full nodes.
```
Comment (maqi) [inline src/capacity/rate_limit.rs:40]: of

File: src/capacity/rate_limit.rs [hunk 2]
```diff
@@ -37,20 +37,20 @@ impl RateLimit {
...
     pub async fn increase_full_node_count(&mut self, node_id: PublicKey) -> Result<()> {
         self.capacity.increase_full_node_count(node_id).await
     }
 
-    ///
-    #[allow(unused)]
-    pub async fn check_network_storage(&self) -> bool {
-        info!("Checking network storage");
+    /// Returns `true` if more storage capacity
+    /// is required, otherwise `false`.
```
Comment (maqi) [inline src/capacity/rate_limit.rs:46]: no need to line break?

**Review by lionel1704 — APPROVED** [review]
Looks good to me

File: src/event_mapping/mod.rs [hunk 3]
```diff
@@ -184,44 +184,23 @@ pub async fn map_routing_event(event: RoutingEvent, network_api: &Network) -> Ma
...
+            ..
+        } => {
+            //self.log_node_counts().await;
+            let op = if let Some(prev_name) = previous_name {
+                trace!("A relocated node has joined the section.");
+                // Switch joins_allowed off a new adult joining.
+                NodeDuty::SetNodeJoinsAllowed(false)
+            } else {
+                //trace!("New node has just joined the network and is a fresh node.",);
+                NodeDuty::NoOp
```
Comment (dirvine) [inline src/event_mapping/mod.rs:200]: Does this also need to be `NodeDuty::SetNodeJoinsAllowed(false)` unless we are in section null (which routing already handles as it adds till section null is recommended section size)

Comment (oetyng) [reply src/event_mapping/mod.rs:200]: Yeah maybe 🤔

File: src/transfers/mod.rs [hunk 4]
```diff
@@ -140,11 +141,14 @@ impl Transfers {
...
-        })
-        //let mut ops = vec![];
-        //ops.push();
-        //ops.push(Ok(ElderDuty::SwitchNodeJoin(self.rate_limit.check_network_storage().await).into()));
-        //ops
+        });
+
+        ops.push(response);
+        ops.push(NodeDuty::SetNodeJoinsAllowed(
+            self.rate_limit.more_nodes_required().await,
```
Comment (maqi) [inline src/transfers/mod.rs:148]: is this the place that will set the flag back to true ?

**Review by dirvine — DISMISSED** [review]

File: src/transfers/mod.rs (hunk 4 above)
Comment (oetyng) [reply src/transfers/mod.rs:148]: Yes, and would only do so if we are at >50% storage capacity.

Comment (oetyng) [reply src/transfers/mod.rs:148]: Not great for spinning up multiple sections fast..

**Review by dirvine — APPROVED** [review]

**Pull request merged**